### ISSUE TITLE
note() and @example() cannot be used together

### ISSUE THREAD

**Issue by csawyerYumaed** [DRMacIver/hypothesis#243]
hypothesis.errors.InvalidArgument: Cannot make notes outside of build context

```python
@given(user=st.text(min_size=1), service=st.text(min_size=1))
def test_tickets(user, service):
	t=random.choice(('ST', 'LT', 'TG', 'CT'))
	note("ticket type: %s" % t)
       # asserts go here
```

I'm not sure what I'm doing wrong here.. Ideally I'd love for hypothesis to choose one of my choices, but hypothesis.strategies.choice() does not take any arguments, so I don't know how to have it pick.
help :)

**Comment by DRMacIver** [comment]
Hi! For future reference, I've edited your report so that the code formats correctly. You can enclose code with backticks and the language name to get syntax highlighting. See "Syntax highlighting" section of https://guides.github.com/features/mastering-markdown/#syntax

On the random.choice front, you want one of two things:

choices() gives you a function that behaves like random.choice:

```python

@given(user=st.text(min_size=1), service=st.text(min_size=1), choice=st.choices())
def test_tickets(user, service, choice):
    t=choice(('ST', 'LT', 'TG', 'CT'))
```
or sampled\_from gives you a strategy for values from a list:

```python

@given(user=st.text(min_size=1), service=st.text(min_size=1), choice=st.choices(), t=st.sampled_from(('ST', 'LT', 'TG', 'CT')))
def test_tickets(user, service, choice):
    pass
```

You probably want sampled_from.

On the specific issue:

I can't seem to reproduce this. I've run your code and it passes fine. It emits a warning because of the use of random.choice but no other problems. Are you sure this is the actual code that was demonstrating the problem and you haven't accidentally oversimplified it?

**Comment by csawyerYumaed** [comment]
Thanks! find highlighed code here :)
OK. you are correct, it only crashes when I have an `@example` defined.  Here is a full test case, and to be sure, I pulled from github and installed that, to ensure we were using the same release.

Also, YAY for sample_from.  I will definitely use that, instead of random.choice(). I didn't know how to do that! (the docs didn't make it clear)

```python
from hypothesis import given, example, assume, note, strategies as st
import random


@given(user=st.text(min_size=1), service=st.text(min_size=1))
@example(user='loser', service='loser')
def test_tickets(user, service):
	t=random.choice(('ST', 'LT', 'TG', 'CT'))
	note("ticket type: %s" % t)
	assert t in ('ST','LT','TG','CT')
	assert user

if __name__ == '__main__':
    test_tickets()
```

**Comment by DRMacIver** [comment]
Oh! Right. Yes. I see what's going on. Thanks. I'm going to update the ticket title to reflect the actual problem, but should have a fix out soon.

**Comment by DRMacIver** [comment]
This should be fixed in 1.17.1 (just released)
